lib/pure/typetraits.nim

```diff
@@ -158,3 +172,6 @@ when isMainModule:
     doAssert a2 == "int"
     doAssert a3 == "int"
   fun(int)
+
+  var a: seq[int]
+  doAssert typeof(a).genericParams.get(0) is int
```

Comment:
there's already a `block genericParams:` in ttypetraits.nim; can you please move the entire isMainModule block and merge it into ttypetraits.nim? avoids splitting tests (and better for reasons mentioned in other PRs)